### PR TITLE
T11925: allow upload for staging packages

### PR DESCRIPTION
**Pull request by worldemar**
Specify library name and version:  **lib/1.0**

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by worldemar** [comment]
see #270

File: .github/workflows/all.yml [hunk 1]
```diff
@@ -22,28 +24,58 @@ env:
...
+            compiler_version: '16'
+            vs_runtimes: 'MD,MT'
+            conan_txt: 'trassir/win32-vc16/conanfile.txt'
+            conan_profile: 'trassir/win32-vc16/conanprofile.release'
+            build_types: 'Release'
+          - build: 'windows-2019-debug'
+            os: 'windows-2019'
+            compiler: 'vs'
+            compiler_version: '16'
+            vs_runtimes: 'MD,MT'
```
Comment (artem-kamyshev) [inline .github/workflows/all.yml:49]: Для `debug` рантайм тоже нужен отладочный: `MDd, MDt`.

**Review by artem-kamyshev — CHANGES_REQUESTED** [review]

**Review by artem-kamyshev — APPROVED** [review]

**Review by Artalus — CHANGES_REQUESTED** [review]

File: .github/workflows/prospector.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      - 'builder/**'
+      - '.github/workflows/prospector.yml'
+  push:
+    branches:
+      - master
+    paths:
+      - 'recipes/**'
+      - 'trassir/**'
+      - 'builder/**'
+      - '.github/workflows/prospector.yml'
```
Comment (Artalus) [inline .github/workflows/prospector.yml:16]: наверное, recipes и trassir лишнее?

File: .github/workflows/prospector.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+jobs:
+  lint:
+    name: ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
```
Comment (Artalus) [inline .github/workflows/prospector.yml:24]: есть ли смысл гонять проспектор отдельно на разных платформах?

File: builder/conan_tools.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+class ConanfileTxt():
+    def __init__(self, filename, conanfile_required):
+        self.packages = {}
+        if path.isfile(filename):
+            with open(filename) as txt_file:
+                for strline in txt_file.read().splitlines():
+                    self.add_package(strline)
+        elif conanfile_required:
+            raise RuntimeError('File {filename} does not exist')
```
Comment (Artalus) [inline builder/conan_tools.py:125]: .format потерялся

**Review by Artalus — CHANGES_REQUESTED** [review]

**Review by Artalus — APPROVED** [review]

**Pull request merged**